view/frontend/web/js/checkout-button.js

```diff
@@ -82,6 +83,18 @@ function ($, Component, placeAmwalOrder, payAmwalOrder, amwalErrorHandler, urlBu
 
             const addToCartForm = $("#" + formID);
 
+            /**
+             * Check if the 'Add to Cart' button is enabled and displayed.
+             */
+            const checkAddToCartButton = () => {
+                const addToCartButton = addToCartForm.find('button[type="submit"]');
+                if (addToCartButton.length && addToCartButton.is(':visible')) {
+                    self.productButtonContainer.classList.remove('hidden');
+                } else {
+                  self.productButtonContainer.classList.add('hidden');
+                }
+            }
+
```

Comment:
can we add this logic as additional step in form validation in `isProductFormValid` without changing any of the main logic of the component?